interview-questions.md

```diff
@@ -7,44 +7,72 @@ Then, research the question to expand on your answer. Even if you feel you have
+
+Researched answer:
+$ The following steps would be taken.
+$ rails g migration add_foreign_key
+$ rails db:migrate
+$ add_column:animals, :animal_id, :integer
+$ rails:db migrate
 
 2. Which RESTful routes must always be passed params? Why?
 
-  Your answer:
+Your answer: I believe the destroy, update and show routes must always be passed params in order to
+successfully pull information from a linked database. For example, an id passed into one of the previously mentioned
+routes in order to update the chosen item.
 
-  Researched answer:
 
+Researched answer:
 
+The Http verb with the url request and passed params defines the correct route
+ex: delete '/photos/:id
+In the example the HTTP verb/action is delete/destroy, requesting photos and the param of id to specify which
+photo from the database to delete.
 
 3. Name three rails generator commands. What is created by each?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer:
+rails g model - model files to include: migration file, model.rb and model test file
+rails g controller - controller files to include controller_name_controller.rb
+rails g resource
 
+Researched answer:
+controller files generated also include the test.rb file , helpers.rb, view file of controller and scss file of controller.
+rails g resource generates : migration file, models, controller file , views, helpers , scss and route of project
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-action: "GET"    location: /students          
+action: "GET"    location: /students = Index method called to display all items of declared model.
 
-action: "POST"   location: /students       
+action: "POST"   location: /students = Create method called to add data to the db.
 
-action: "GET"    location: /students/new
+action: "GET"    location: /students/new = New method called to display the form for the user to enter data for a new student
 
-action: "GET"    location: /students/2  
+action: "GET"    location: /students/2 = Show method called to list one item from the model in this case the student with an id of 2
 
-action: "GET"    location: /students/2/edit    
+action: "GET"    location: /students/2/edit = Edit method to display a form to edit student information with the id of 2.
 
-action: "PATCH"  location: /students/2      
+action: "PATCH"  location: /students/2 = Update method called and updates the student information in the db
 
-action: "DELETE" location: /students/2      
+action: "DELETE" location: /students/2= Destroy method called and removes the student with an id of 2 from the db
 
+5. As a developer, you are making an application to manage your to do list.
+   Create 10 user stories that will help you get your application started.
+   Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
 
+   USER STORIES:
```

Comment:
These are good starting points for the user experience that helps a developer understand the end goal